src/hooks/metamask.hook.ts

```diff
@@ -60,11 +62,64 @@ export function useMetaMask(): MetaMaskInterface {
     return web3.eth.personal.sign(message, address, '');
   }
 
+  async function addContract(address: string, svgData: string): Promise<boolean> {
+    const tokenContract = new web3.eth.Contract(
+      [
+        {
+          constant: true,
+          inputs: [],
+          name: 'symbol',
+          outputs: [
+            {
+              name: '',
+              type: 'string',
+            },
+          ],
+          payable: false,
+          stateMutability: 'view',
+          type: 'function',
+        },
+        {
+          constant: true,
+          inputs: [],
+          name: 'decimals',
+          outputs: [
+            {
+              name: '',
+              type: 'uint8',
+            },
+          ],
+          payable: false,
+          stateMutability: 'view',
+          type: 'function',
+        },
+      ],
+      address,
+    );
```

Comment:
What is this fancy thing? 😆

Reply:
if i would know. AFAIU it tells which data should be built inside of the contract. like a very rich builder pattern